src/huggingface_hub/hf_api.py

```diff
@@ -8261,6 +8277,173 @@ def grant_access(
         hf_raise_for_status(response)
         return response.json()
 
+    ###################
+    # Manage webhooks #
+    ###################
+
+    @validate_hf_hub_args
+    def get_webhook(self, webhook_id: str, token: Optional[str] = None) -> Dict:
+        """Get a webhook by its id."""
+        response = get_session().get(
+            f"{ENDPOINT}/api/settings/webhooks/{webhook_id}",
+            headers=self._build_hf_headers(token=token),
+        )
+        hf_raise_for_status(response)
+        return response.json()
+
+    @validate_hf_hub_args
+    def list_webhooks(self, token: Optional[str] = None) -> List[Dict]:
+        """List all configured webhooks."""
+        response = get_session().get(
+            f"{ENDPOINT}/api/settings/webhooks",
+            headers=self._build_hf_headers(token=token),
+        )
+        hf_raise_for_status(response)
+        return response.json()
+
+    @validate_hf_hub_args
+    def create_webhook(
+        self,
+        watched: List[WatchedItem],
+        url: str,
+        domains: List[WEBHOOK_DOMAIN_T],
+        secret: Optional[str] = None,
+        token: Optional[str] = None,
+    ) -> Dict:
+        """Create a new webhook.
+
+        Args:
+            watched (List[WatchedItem]):
+                List of items to watch. It an be users, orgs, models, datasets or spaces.
+                See `WatchedItem` for more details.
+            url (str):
+                URL to send the payload to.
+            domains (List[Literal["repo", "discussion"]]):
+                List of domains to watch. It can be "repo", "discussion" or both.
+            secret (`str`, *optional*):
+                A secret to sign the payload with.
+            token (`str`, *optional*):
+                A valid authentication token (see https://huggingface.co/settings/token).
+
+        Returns:
+            dict: The created webhook.
```

Comment:
```suggestion
            [`Webhook`]: The created webhook.
```

(nit) same for returns value that can be referenced (once `Webhook` dataclass will be defined)